### PR TITLE
servoshell: add keyboard shortcuts for switching tabs

### PR DESCRIPTION
**Pull request by Melchizedek6809**
This PR adds Cmd/Ctrl + [1-9] for selecting a particular tab and Ctrl + PageUp/Down for selecting the previous/next tab. Thought this might be convenient since they are present in all major browsers (only safari seems to use different keys for switching to the previous/next tab)

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes do not require tests because UI

### CODE REVIEW

File: ports/servoshell/desktop/webview.rs [hunk 1]
```diff
@@ -414,6 +414,29 @@ where
         self.shutdown_requested
     }
 
+    fn focus_webview_by_index(&self, index: usize) -> Option<EmbedderEvent> {
+        let webviews = &self.webviews();
+        if index < webviews.len() {
```
Comment (mukilan) [inline ports/servoshell/desktop/webview.rs:419]: I think we can simplify this if/else using the `?` operator

```
Some(EmbedderEvent::FocusWebView(webviews.get(index)?.0))
```

File: ports/servoshell/desktop/webview.rs [hunk 2]
```diff
@@ -414,6 +414,29 @@ where
...
+            Some(EmbedderEvent::FocusWebView(webviews[index].0))
+        } else {
+            None
+        }
+    }
+
+    fn get_focused_webview_index(&self) -> Option<usize> {
+        let focused_id = match self.focused_webview_id {
```
Comment (mukilan) [inline ports/servoshell/desktop/webview.rs:427]: The match can be simplified to
```
let focused_id = self.focused_webview_id?;
```

File: ports/servoshell/desktop/webview.rs [hunk 3]
```diff
@@ -414,6 +414,29 @@ where
...
+            Some(id) => id,
+            None => return None,
+        };
+        let webviews = &self.webviews();
+        for i in 0..webviews.len() {
```
Comment (mukilan) [inline ports/servoshell/desktop/webview.rs:432]: We can replace the for loop with

```
webviews.iter().position(|webview| webview.0 == focused_id)
```

**Comment by Melchizedek6809** [comment]
@mukilan Thank you for the review! Incorporated your advice and the code is quite a bit shorter/cleaner now

**Review by mukilan — APPROVED** [review]
Thank you for the contribution!

**Pull request merged**